src/components/NestedEditComponent.tsx

```diff
@@ -307,6 +307,10 @@ export class NestedEditComponent extends React.Component<InlineComponentProps, a
             if (field.hasOwnProperty("defaultValueFn") && (!props.currentModel || !props.currentModel.hasOwnProperty(field.field))) {
                 defaultValue[field.field] = field.defaultValueFn()
             }
+
```

Comment:
like `defaultValueFn`, we can have `valueFn` where the component can decide what the value needs to be.
props will be passed to this function so all custom functionalities can be achieved.